### PR TITLE
Feature/mob 4402 input cells for checkout and 4456 car details 

### PR DESCRIPTION
**Pull request by BartlomiejSopalaKarhoo**
## JIRA
[MOB-4402](https://karhoo.atlassian.net/browse/MOB-4402)
[MOB-4456](https://karhoo.atlassian.net/browse/MOB-4456)

## DESCRIPTION
This PR contains 2 tickets: 4402 and 4456

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

[MOB-4402]: https://karhoo.atlassian.net/browse/MOB-4402?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MOB-4456]: https://karhoo.atlassian.net/browse/MOB-4456?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                        .frame(
+                            maxWidth: UIConstants.Dimension.Icon.medium,
+                            maxHeight: UIConstants.Dimension.Icon.medium
+                        ).scaledToFill()
+                        
+                    Text(viewModel.fleetName)
+                        .font(Font(KarhooUI.fonts.captionSemibold()))
+                        .foregroundColor(KarhooUI.colors.text.getColor())
+                    
+
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift:38]: Empty vertical whitespaces

File: KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+struct VehicleDetailsCardViewModel {
+    let title: String
+    let passengerCapacity: Int
+    let luggageCapacity: Int
+    let fleetName: String
+    let carIconUrl: String
+    let fleetIconUrl: String
+}
+
+struct NewVehicleCapacity: View {
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift:74]: If it's a view needed only for the feature, maybe set it as private? If not, move it to designed file

File: KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutPresenter.swift [hunk 3]
```diff
@@ -139,4 +175,25 @@ final class KarhooNewCheckoutViewModel: ObservableObject {
         default: return false
         }
     }
+    
+    private func shouldShowTrainNumberCell() -> Bool {
+        if journeyDetails.isScheduled,
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutPresenter.swift:180]:
```
return journeyDetails.isScheduled &&
           quote.fleet.capability.compactMap({ FleetCapabilities(rawValue: $0) }).contains(.trainTracking) &&
           journeyDetails.originLocationDetails?.details.type == .trainStation 
```

File: KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutPresenter.swift [hunk 4]
```diff
@@ -139,4 +175,25 @@ final class KarhooNewCheckoutViewModel: ObservableObject {
...
+           quote.fleet.capability.compactMap({ FleetCapabilities(rawValue: $0) }).contains(.trainTracking),
+           journeyDetails.originLocationDetails?.details.type == .trainStation {
+            return true
+        } else {
+            return false
+        }
+    }
+    
+    private func shouldShowFlightNumberCell() -> Bool {
+        if journeyDetails.isScheduled,
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutPresenter.swift:190]:
```
private func shouldShowFlightNumberCell() -> Bool {
    journeyDetails.isScheduled &&
    quote.fleet.capability.compact M ap({ FleetCapabilities(rawValue: $0) }).contains(.flightTracking) &&
    journeyDetails.originLocationDetails?.details.type == .airport
}
```

File: KarhooUISDKTestUtils/Mocks/Common/MockDataFormatterType.swift [hunk 5]
```diff
@@ -63,10 +63,12 @@ public class MockDateFormatterType: DateFormatterType {
...
-    public func display(_ date: Date?, dateStyle: DateFormatter.Style, timeStyle: DateFormatter.Style) -> String {
-        fullTimeSet = date
-        return customDateFormatToReturn
+    
+    public func display(
+        _ date: Date?,
+        dateStyle: DateFormatter.Style,
+        timeStyle: DateFormatter.Style
+    ) -> String {
+        TestUtil.getRandomString()
```
Comment (AleksanderWedrychowskiKarhoo) [inline KarhooUISDKTestUtils/Mocks/Common/MockDataFormatterType.swift:72]: You can not return random string, cuz it won't be working with snapshot tests

Comment (BartlomiejSopalaKarhoo) [reply KarhooUISDKTestUtils/Mocks/Common/MockDataFormatterType.swift:72]: If you take a look at this class every "display" method return random string. I can hardcoded some value, but rather we should create ticket and check why/how it worked for now.

**Comment by BartlomiejSopalaKarhoo** [comment]
Changes from this this PR was merged in another 2 PR's.

**Pull request closed without merging**